crates/router/src/connector/stancer.rs

```diff
@@ -0,0 +1,504 @@
+pub mod transformers;
+
+use std::fmt::Debug;
+use error_stack::{ResultExt, IntoReport};
+use masking::ExposeInterface;
+
+use crate::{
+    configs::settings,
+    utils::{self, BytesExt},
+    core::{
+        errors::{self, CustomResult},
+    },
+    headers, services::{self, ConnectorIntegration, ConnectorValidation, request::{self, Mask}},
+    types::{
+        self,
+        api::{self, ConnectorCommon, ConnectorCommonExt},
+        ErrorResponse, Response,
+    }
+};
+
+
+use transformers as stancer;
+
+#[derive(Debug, Clone)]
+pub struct Stancer;
+
+impl api::Payment for Stancer {}
+impl api::PaymentSession for Stancer {}
+impl api::ConnectorAccessToken for Stancer {}
+impl api::MandateSetup for Stancer {}
+impl api::PaymentAuthorize for Stancer {}
+impl api::PaymentSync for Stancer {}
+impl api::PaymentCapture for Stancer {}
+impl api::PaymentVoid for Stancer {}
+impl api::Refund for Stancer {}
+impl api::RefundExecute for Stancer {}
+impl api::RefundSync for Stancer {}
+impl api::PaymentToken for Stancer {}
+
+impl
+    ConnectorIntegration<
+        api::PaymentMethodToken,
+        types::PaymentMethodTokenizationData,
+        types::PaymentsResponseData,
+    > for Stancer
+{
+    // Not Implemented (R)
+}
+
+impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Stancer
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,{
+    fn build_headers(
+        &self,
+        req: &types::RouterData<Flow, Request, Response>,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
+        let mut header = vec![(
+            headers::CONTENT_TYPE.to_string(),
+            self.get_content_type().to_string().into(),
+        )];
+        let mut api_key = self.get_auth_header(&req.connector_auth_type)?;
+        header.append(&mut api_key);
+        Ok(header)
+    }
+}
+
+impl ConnectorCommon for Stancer {
+    fn id(&self) -> &'static str {
+        "stancer"
+    }
+
+    fn get_currency_unit(&self) -> api::CurrencyUnit {
+        api::CurrencyUnit::Minor
+    }
+
+    fn common_get_content_type(&self) -> &'static str {
+        "application/json"
+    }
+
+    fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
+        connectors.stancer.base_url.as_ref()
+    }
+
+    fn get_auth_header(
+        &self,
+        auth_type: &types::ConnectorAuthType,
+    ) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
+        let auth = stancer::StancerAuthType::try_from(auth_type)
+            .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+        Ok(vec![(
+            headers::AUTHORIZATION.to_string(),
+            format!("Bearer {}", auth.api_key.expose()).into_masked(),
```

Comment:
```suggestion
            format!("Basic {}", auth.api_key.expose()).into_masked(),
```
According to documentation, stancer uses `Basic` Authentication.